bigquery_etl/dryrun.py

```diff
@@ -350,8 +349,15 @@ def dry_run_result(self):
                 job = self.client.query(sql, job_config=job_config)
                 try:
                     dataset_labels = self.client.get_dataset(job.default_dataset).labels
-                except PermissionDenied:
-                    dataset_labels = []
+                except Exception as e:
+                    # Most users do not have bigquery.datasets.get permission in
+                    # moz-fx-data-shared-prod
+                    # This should not prevent the dry run from running since the dataset
+                    # labels are usually not required
+                    if "Permission bigquery.datasets.get denied on dataset" in str(e):
+                        dataset_labels = []
+                    else:
+                        raise e
```

Comment:
I was able to reproduce this in my sandbox project by creating a service account with limited permissions. Turns out here only a general `Exception` is thrown instead of a `PermissionDenied` exception